### DISCUSSION TITLE
Bootable Linux (grub) found but will not boot it

### DISCUSSION THREAD

**Discussion by britdavis** [CloverHackyColor/CloverBootloader#519] [Q&A] [answered]
Hello,
I'm running a Dell Poweredge 720 with an NVMe drive on a PCIe card/bridge. The newest BIOS for the 720 does not support NVMe boot, so I'm trying Clover.
I found some good writeups on how to make this work but the solution involves putting the volume GUID in the config.plist. My issue is that this volume is formatted as MBR (vs GPT) and thus does not have a GUID but rather a volume ID:

```
7:926  0:011  - [50]: Volume: PciRoot(0x2)\Pci(0x1,0x0)\Pci(0x0,0x0)\NVMe(0x1,85-9E-84-25-68-B7-26-00)
7:926  0:000          Result of bootcode detection: bootable Linux (grub,linux)
7:926  0:000  - [51]: Volume: PciRoot(0x2)\Pci(0x1,0x0)\Pci(0x0,0x0)\NVMe(0x1,85-9E-84-25-68-B7-26-00)\HD(1,MBR,0x000E1F53,0x800,0x1BF23800)
```
None of the two default (or three if you reveal the hidden one) finds that bootable partition.
I've attached both my preboot.log and config.plist.
Any suggestions would be very much appreciated
[preboot.log](https://github.com/CloverHackyColor/CloverBootloader/files/8879225/preboot.log)
[config.plist.txt](https://github.com/CloverHackyColor/CloverBootloader/files/8879231/config.plist.txt)
!

**Comment by Morbius01** [comment, marked as answer]
FYI, I have successfully installed several OSs (Windows and Linux) through Clover on a platform similar to yours (non bootable NVMe SSD on a PCIe adapter) as you can see [here](https://github.com/CloverHackyColor/CloverBootloader/discussions/491) :

Clover works fine, as long as you strictly follow the guidelines when installing it.

If you are interested I can help you in describing for you with details the steps I followed, so that you could replicate the same process.

Installing Clover requires 2 steps.
1. Creating a Clover key
2. Installing the OS on your non bootable drive through Clover

So, the first question I have for you is : have you successfully created a Clover key (Step 1) ?

**Reply by britdavis** [reply]
Also, does this mean that an OS that is already installed on a drive that is not currently bootable (because of BIOS limitations) that Clover will not be able to boot it at all unless the OS had been installed through clover?

**Reply by britdavis** [reply]
Hi Morbius01, sorry for my delay and thank you for you response.
Not trying to seem dumb, but key is my bootable USB flash drive? If so then yes.
I think what I must have missed was that I was supposed to install the OS that clover is to later boot using clover.
I'll give that a try.
:-)

**Reply by Morbius01** [reply]
Hi britdavis.
Just saw your message.

> Not trying to seem dumb, but key is my bootable USB flash drive?

You guessed it right. What I mean by 'Clover key' is a USB drive on which you have installed Clover, and which is now bootable.


> I think what I must have missed was that I was supposed to install the OS that clover is to later boot using clover. I'll give that a try. :-)

Absolutely.


> Also, does this mean that an OS that is already installed on a drive that is not currently bootable (because of BIOS limitations) that Clover will not be able to boot it at all unless the OS had been installed through clover?

I can't be absolutely sure and I can't rule out the possibility of making Clover to boot on an OS previously installed.
However, I have never done so. I wouldn't know how to do it, and I wouldn't waste my time trying to do it. And, of course, if you were deciding to go this way, I wouldn't be able to help you.

So, my understanding is that you have now succeeded in creating a Clover key that works, meaning that you can boot through it and, at the end, you get the Clover menu on your screen !   Congrats for achieving Step 1 !

Can I ask you what OS(s) you are trying to install through Clover ?
And, have you gone any further and perhaps succeeded in installing it (them) and boot through Clover ?
